Signing/node-express/src/app.ts

```diff
@@ -158,17 +162,23 @@ app.post(
 // Listen to incoming events
 app.post('/webhook', async (req: Request, res: Response) => {
   const { event, signatureOrderId } = req.body;
-  const order = await getSignatureOrder(signatureOrderId);
   let orderCanClose = false;
 
-  if (event) {
-    console.log(
-      `Received event ${event} for order ${signatureOrderId}`,
-      req.body
-    );
+  if (event === "WEBHOOK_VALIDATION") {
```

Comment:
Ah, this was what was causing it to hang?

Reply:
Should probably add a try/catch that captures any errors in handling and responds with an error code.

Reply:
> this was what was causing it to hang?

Worse, the line below was creating an order right away, without the orderID, before validation. 

`const order = await getSignatureOrder(signatureOrderId);`

Reply:
@nmoskaleva Aye yes, that's what I meant, I assume that `const order = await getSignatureOrder(signatureOrderId);` would fail in some way, or some code dependent on it would fail.

Uncaught exceptions don't result in an error response in express I think, unless you add explicit try/catch middleware.